### PR TITLE
MAID-2527 refactor/mdata_entries: update mdata_entries_for_each to md…

### PR DESCRIPTION
**Pull request by hunterlester**
…ata_list_entries change in safe_client_libs

In order to test this PR properly you'll need to build https://github.com/maidsafe/safe_client_libs/tree/master/safe_app locally with `cargo build --release --features=use-mock-routing` and copy the resulting `safe_app.dll` in `target/release` into `safe_app_nodejs/src/native/mock`.

CI will fail until latest `safe_client_libs` libraries are published.

### CODE REVIEW

**Comment by bochaco** [comment]
@hunterlester , you can use the commit number of the safe_client_libs in our package.json and it will fetch it, e.g. 
```
    "safe_app": {
      "mirror": "https://s3.eu-west-2.amazonaws.com/safe-client-libs",
      "version": "0e99724",
      ...
```

File: src/native/_mutable.js [hunk 1]
```diff
@@ -342,7 +348,26 @@ module.exports = {
...
+      const len = args[1];
+      const entriesList = [];
+      if (len > 0) {
+        let arrPtr = ref.reinterpret(ptr, MDataEntry.size * len);
+        let arr = MDataEntriesArray(arrPtr);
+        for (let i = 0; i < len ; i++) {
+          const currEntry = arr[i];
+          const keyStr = ref.reinterpret(currEntry.key.val_ptr, currEntry.key.val_len, 0);
+          const valueStr = ref.reinterpret(currEntry.value.content_ptr, currEntry.value.content_len, 0);
+	  const entryObject = {
```
Comment (bochaco) [inline src/native/_mutable.js]: There is a minor formatting issue here

File: test/immutable.js [hunk 2]
```diff
@@ -151,8 +151,9 @@ describe('Immutable Data', () => {
         )
       .then(() => app.mutableData.newPublic(testXorName, TYPE_TAG))
       .then((md) => md.getEntries())
-      .then((entries) => entries.forEach((key, value) => {
-        app.immutableData.fetch(value.buf)
+      .then((entries) => entries.listEntries())
+      .then((entriesArray) => entriesArray.forEach((entry) => {
```
Comment (bochaco) [inline test/immutable.js]: I think we are missing something here, shouldn't we be doing all the `fetch`'s within a Promise.all?

File: test/error_const.js [hunk 3]
```diff
@@ -100,4 +99,9 @@ describe('Error Constants', () => {
     should(error.INVALID_PERM.msg('theReason')).startWith('theReason');
     should.exist(error.INVALID_PERM);
   });
+
+  it('INVALID_SEC_KEY', () => {
+    should(new RegExp(32).test(error.INVALID_SEC_KEY.msg(32))).be.true();
```
[the remark this answers is not in the document]
Comment (hunterlester) [reply test/error_const.js]: This is validating that the string constructed by the `msg` function is composed with the given parameter.

**Pull request merged**